xmtp_mls/src/groups/mls_sync.rs

```diff
@@ -1381,8 +1396,7 @@ where
                             installation_id = %self.client.installation_id(),
                             "Skipping intent because no publish data returned"
                         );
-                        let deleter: &dyn Delete<StoredGroupIntent, Key = i32> = provider.conn_ref();
-                        deleter.delete(intent.id)?;
+                        provider.conn_ref().set_group_intent_processed(intent.id)?
```

Comment:
The one thing we should keep in mind is that the old workflow was idempotent, and the new one will return an error if the intent was already deleted.

How confident are we that this won't cause issues with concurrency/retries?

Reply:
Great question, as far as I checked the code we have two places that we get the intents to process them and we already handle the processed case there.